src/gui/dialogs/unit_recall.cpp

```diff
@@ -235,6 +240,36 @@ void unit_recall::pre_show(window& window)
 	list_item_clicked(window);
 }
 
+void unit_recall::rename_unit(window& window)
+{
+	listbox& list = find_widget<listbox>(&window, "recall_list", false);
+
+	const int index = list.get_selected_row();
+	const unit& selected_unit = *recall_list_[index].get();
+
+	std::string name = selected_unit.name();
+	const std::string dialog_title(N_("Rename Unit"));
+	const std::string dialog_label(N_("Name:"));
+
+	if(gui2::dialogs::edit_text::execute(dialog_title, dialog_label, name)) {
+		unit_ptr unit_to_rename = team_.recall_list().find_if_matches_id(selected_unit.id());
```

Comment:
Not sure why you create a `unit_ptr` here. `rename` is a member of `unit`, and you already have a unit reference above (just make it non-const).

Reply:
Hmm... right. I had to missed something when messing with code.